src/layers/data_type_layer.cpp

```diff
@@ -781,28 +812,24 @@ void data_type_layer<InputTensorDataType, OutputTensorDataType>::
 fp_setup_inputs(El::Int mini_batch_size) {
   if (get_num_parents() < 1) { return; }
 
-  // Determine distributed matrix alignment
-  const auto& alignment_dist = get_parent_layer().get_activations(*this).DistData();
```

Comment:
Are new no longer ensuring that the matrices have the same alignment?

Reply:
Yep, non-aligned matrices will now trigger communication.